app/src/main/java/org/kiwix/kiwixmobile/main/KiwixWebViewClient.java

```diff
@@ -108,9 +114,21 @@ private void inflateHomeView(WebView view) {
     LayoutInflater inflater = LayoutInflater.from(view.getContext());
     home = inflater.inflate(R.layout.content_main, view, false);
     callback.setHomePage(home);
+
     if (sharedPreferenceUtil.nightMode()) {
       ImageView cardImage = home.findViewById(R.id.content_main_card_image);
-      cardImage.setImageResource(R.drawable.ic_home_kiwix_banner_night);
+      ImageView sideImage = home.findViewById(R.id.content_side_image);
+      AppCompatButton downloadButton = home.findViewById(R.id.content_main_card_download_button);
+      ConstraintLayout constraintLayout = home.findViewById(R.id.constraint_main);
+      constraintLayout.setBackgroundResource(R.drawable.back_cover_night);
```

Comment:
A very nice follow-up. I think you can make use of this answer https://stackoverflow.com/a/13735919/7792129 to change the image based on the theme. Although we were doing it dynamically we should change it. This will move the UI implementation to the xml where it should be.

Reply:
@abdulwd I tried this but it's not working. The app crashes when I did that.
Also, I don't think it's worth adding 4 lines when we are getting the same desired result in 1 line.